src/main/java/terminator/Terminator.java

```diff
@@ -0,0 +1,128 @@
+package terminator;
+
+import java.util.Scanner;
+
+import terminator.command.Command;
+import terminator.command.CommandParser;
+import terminator.command.DeadlineCommand;
+import terminator.command.DeleteCommand;
+import terminator.command.EventCommand;
+import terminator.command.TerminatorException;
+import terminator.command.TodoCommand;
+import terminator.task.Storage;
+import terminator.task.TaskList;
+
+/**
+ * Main class for the chatbot application.
+ */
+public class Terminator {
+
+    private final TaskList taskList;
+
+    private final CommandParser parser;
+
+    private final TerminalUi terminalUi;
+
+    private final Storage storage;
+
+    /**
+     * Creates a new instance of the Terminator object, which is the main entry point of the application.
+     */
+    public Terminator() {
+        this.taskList = new TaskList();
+        this.parser = new CommandParser();
+        this.terminalUi = new TerminalUi();
+        this.storage = new Storage();
+    }
+
+    /**
+     * Runs the chatbot application, which first attempts to read data from the local disk to populate
+     * the task list. <br/><br/>
+     *
+     * If the data was read successfully, initiate the main event loop, which can
+     * handle commands from the user through the command line. <br/><br/>
+     *
+     * The application exits when the user enters {@code bye} into the command line.
+     */
+    private void runTerminalUi() {
+        // Load data from storage
+        try {
+            this.storage.loadDataFromFile(this.taskList);
```

Comment:
nit: can remove unnecessary use of this keyword. Should only use this keyword if there is ambiguity in code.